.github/scripts/release.py

```diff
@@ -60,40 +60,54 @@ def get_package_sha256(pkg):
+        data = json.load(f)
+    return data
+
 def get_platform_data_struct_copy(pkg_index):
-    return copy.deepcopy(pkg_index['packages'][0]['platforms'][0])
+    return copy.deepcopy(pkg_index['packages'][0]['platforms'])
 
-def set_new_platform_data_fields(platform_data, pkg_name, version):
+def set_new_platform_data_fields(platform_data_index, pkg_name, version, repository):
     semver = strip_prefix_from_version(version)
+    platform_data = platform_data_index['packages'][0]['platforms'][0]
     platform_data['version'] = str(semver)
     archive_file_name = str(pkg_name) + ".zip"
     platform_data['archiveFileName'] = archive_file_name
-    platform_data['url'] = "https://github.com/Infineon/XMC-for-Arduino/releases/download/" + str(version) + "/" + str(archive_file_name)
+    platform_data['url'] = "https://github.com/" + str(repository) + "/releases/download/" + str(version) + "/" + str(archive_file_name)
     platform_data['checksum'] ="SHA-256:" + str(get_package_sha256(os.path.join(pkg_assets_build_path, archive_file_name)))
     platform_data['size'] = str(get_package_size(os.path.join(pkg_assets_build_path, archive_file_name)))
 
-def add_new_platform_to_package_index(pkg_index, new_platform):    
-    pkg_index['packages'][0]['platforms'].insert(0, new_platform)
+def add_platform_to_package_index(pkg_index, platform):    
+    pkg_index['packages'][0]['platforms'].extend(platform)
 
 def make_package_index_file(pkg_index):
     pkg_index_json_obj = json.dumps(pkg_index, indent=2)
     pkg_index_w_path = os.path.join(pkg_assets_build_path, "package_infineon_index.json")
     with open(pkg_index_w_path, "w") as pkg_file:
         pkg_file.write(pkg_index_json_obj)
 
-def build_package_index_json(pkg_name, version):
-    package_index = get_latest_package_index_json()
-    new_platform_data = get_platform_data_struct_copy(package_index)
-    set_new_platform_data_fields(new_platform_data, pkg_name, version)
-    add_new_platform_to_package_index(package_index, new_platform_data)
-    make_package_index_file(package_index)
-
-def build_release_assets(version):
+def build_package_index_json(pkg_name, version, repository):
+    # get online package index json
+    latest_package_index = get_latest_package_index_json() 
+    # get local package index template
+    local_package_index = get_local_package_index_json() 
+    # set data field in local template for newest package
+    set_new_platform_data_fields(local_package_index, pkg_name, version, repository)
+    # get old package array
+    old_platform_data = get_platform_data_struct_copy(latest_package_index)
+    # append to local package index
+    add_platform_to_package_index(local_package_index, old_platform_data)
+    make_package_index_file(local_package_index)
+
+def build_release_assets(version, repository):
+    if os.path.exists(pkg_assets_build_path):
+        os.system("rm -rf "+pkg_assets_build_path)
```

Comment:
Why using rm to delete the package? 
This will make this script only unix compatible.

Reply:
I was testing it in wsl because the runner is also unix...
I can also remove it. This is easy for local test, but is most likely not needed for workflow  ( the path is clean

Reply:
You don´t need to remove it, but instead use python apis to remove filesystem folders and files. That way we keep it cross platform.